lib/src/actions/package/providers/zypper.rs

```diff
@@ -0,0 +1,79 @@
+use super::PackageProvider;
+use crate::actions::package::{repository::PackageRepository, PackageVariant};
+use crate::atoms::command::Exec;
+use crate::steps::Step;
+use serde::{Deserialize, Serialize};
+use tracing::warn;
+use which::which;
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct Zypper {}
+
+impl PackageProvider for Zypper {
+    fn name(&self) -> &str {
+        "Zypper"
+    }
+
+    fn available(&self) -> bool {
+        match which("zypper") {
+            Ok(_) => true,
+            Err(_) => {
+                warn!(message = "zypper not availiable");
```

Comment:
minor nit `available`

Reply:
Spelling correction has been taken care of, good catch! Even as a native English speaker, this is one of those words that get me every time.